src/nvim/menu.c

```diff
@@ -1301,11 +1301,15 @@ void ex_emenu(exarg_T *eap)
 
   /* Found the menu, so execute.
    * Use the Insert mode entry when returning to Insert mode. */
-  if (restart_edit
+  if (
+      ((State & INSERT) || restart_edit)
```

Comment:
Why don't you use `get_real_state()` here as well or why is it only necessary below?
Why isn't `((State & INSERT) || restart_edit)` on the same line as the if?